font-codegen/src/record.rs

```diff
@@ -209,13 +209,31 @@ pub(crate) fn generate_compile_impl(
         }
     });
 
+    let can_derive_default = fields.can_derive_default()?;
+    let maybe_derive_default = can_derive_default.then(|| quote!(Default));
+    let maybe_custom_default = (!can_derive_default).then(|| {
+        let default_field_inits = fields.iter_compile_default_inits();
+        let default_impl_params = generic_param.map(|t| quote! { <#t: Default> });
+        quote! {
+        impl #default_impl_params Default for #name <#generic_param> {
+            fn default() -> Self {
+                Self {
+                    #( #default_field_inits, )*
+                }
+            }
+        }
+        }
+    });
+
     Ok(quote! {
         #( #docs )*
-        #[derive(Clone, Debug)]
+        #[derive(Clone, Debug, #maybe_derive_default)]
```

Comment:
Optional: I dislike the trailing comma when maybe_derive_default is empty. Imagine we add a few more maybe_whatever's, you could get stuff like `#[derivce(Clone, Debug, , , )]`. That would lead me to put the , into maybe_derive_default.

Reply:
the trialing comma will be removed when we run rustfmt on the generated code. Does that change your thinking?

Reply:
maybe we should have a trailing comma in `#maybe_derive_default` as well?